### ISSUE TITLE
several minor problems

### ISSUE THREAD

**Issue by pmattern** [aericson/aarchup#8]
This issue is meant to summarize a couple of problems which should be improved, imo, but which don't justify a bug report on their own.
All findings made with recent AUR package aarchup.

*icons*
The build procedure creates archlogo.svg but skips archlogo.png. If this is intentional (which I don't believe), aarchupstartup.example should be modified as it's referencing archlogo.png.
I think archlogo.{png,svg} should be placed in /usr/share/icons/hicolor/{32x32,scalable}/apps. This would be according to the [Icon Theme Specification](http://www.freedesktop.org/wiki/Specifications/icon-theme-spec) and hence make the icons available to other applications as well.

*systemd*
According to systemd documentation, see e. g. Section "Unit Load Path" in [man (5) systemd.unit](http://www.freedesktop.org/software/systemd/man/systemd.unit.html) unit files provided by packages should go to /usr/lib/systemd/system. Folder /etc/systemd/system is for local configuration adjustments only, e. g. by copying units from /usr or by using those snippets in subfolders <service>.service.d. So aarchup.{service,timer} should be moved accordingly.
As a side effect line backup in PKGBUILD wouldn't have to cover aarchup.timer any longer.
Regarding systemd.timer I think 15min is a bit much. Normally one wants to be informed as soon as possible after booting. So I'd suggest to reduce to 5min or something.

*noxaarchup, aarchupstartup.example*
noxaarchup is invoking aarchup without --icon. Would be nicer to have it set, imo.
Also, I think both noxaarchup and aarchupstartup.example should set --aur by default as using this switch when cower is lacking seems to cause nothing but an error message.

*messages*
Given Summary starts by "New updates..." setting "Application Name" to "New Updates" doesn't seem like a good choice to me, in particular with regards to notificationd implementations which cite both items in their pop-ups like the one from LXQt. Why not set 'Application Name' just to 'aarchup'?
If updates in official repos are available both old and new version are displayed, regarding AUR only the latter. This should be consistent.

**Comment by aericson** [comment]
First, thanks a lot for your input.

*icons:*
The .png problem should had been fixed, there has been some problems with packaging. I've fixed it now.
About the icon locations you seem to be right, I'll address it.

*systemd:*
I will look into moving them, but why wouldn't the backup line be needed? They are there so that when I change them in future versions, user's changes won't be overwritten (a pacnew will be created).
I also have to disagree with the 5 minutes. It's too low. In case the user is busy, to be notified every 5 minutes is just too annoying. Plus, it's easy enough to change the default.

*noxaarchup, aarchupstartup.example:*
I don't see a reason not to have --icon by default, I'll address it.
I don't think the --aur should be there by default, it's not semantically correct to have it set without cower being installed. Also, aarchup was always meant to be used with pacman with no aur support. This was added as an extra feature and requires an external tool. Then again, it's easy enough to change the default.

*messages:*
Application name should change, yes.
Long time ago, back when this was first released, pacman -Qu didn't use to show the new version of the packages. Hence there is a '+' at the end of each line. In the end it used to show something like:
curl 7.40.0-1+
I never really noticed it to be honest.

Thanks one more time :)

**Comment by pmattern** [comment]
*systemd*
\*.pacnew files get created if files *provided by packages* are modified. But regarding systemd the common habit is to place system-wide units in /usr and handle local configuration changes by different kinds of files that get placed in /etc *additionally* and hence do *not* belong to packages. So no need to backup systemd files in /etc.
You may have a look at packages in the official repos - they do state all kinds of configuration files in /etc in PKBUILD's "backup" line but never systemd units.
The proposal was to modify OnBootSec, not OnUnitActiveSec. Aside from this it's of course right that modifying those values locally is easy.

*messages*
Displaying both current and new versions was introduced in pacman 4.2.0, see [NEWS](https://projects.archlinux.org/pacman.git/tree/NEWS) ("- both current and new...").
Regarding AUR it seems a bit odd to me that aarchup is displaying the new version only as `cower --update` is displaying both versions as well.

**Comment by aericson** [comment]
There is already a configuration file in /etc/aarchup.conf I will look into unifying all configuration to it.
I will change the OnBootSec also. Still, /etc/aarchup.conf would need to be in backup array in PKGBUILD. From [docs](https://wiki.archlinux.org/index.php/Pacnew_and_Pacsave_files):

> For each `backup` file in a package being upgraded, pacman cross-compares three md5sums
> ...
> pacman saves the new version with a .pacnew extension and leaves the modified version untouched.


For the messages, I will change it. AUR updates were only showing new version to keep output similar. But since pacman changed, no need to keep it like that. I guess I forgot to state that I was gonna change that on my previous message.

One more time, thanks for the feedback and the detailed descriptions.

**Comment by pmattern** [comment]
> Still, /etc/aarchup.conf would need to be in backup array in PKGBUILD. 

But that's totally consistent with what I've been saying: any configuration file in /etc that's part of a package should be covered by "backup".
If /etc/aarchup.conf was delivered by the package (which doesn't apply so far) then of course it would have to be covered by "backup".
systemd is an exception in that its configuration files in /etc are normally never provided by packages but result from copying the ones from /usr, adding snippets or whatever.

**Comment by aericson** [comment]
I agree with you, that would be ideal. No idea when I will get around to do that though.
